examples/Reranking_search_results_with_cross-encoders.ipynb

```diff
@@ -0,0 +1,805 @@
+    "Relevant to the query?\n",
+    "Assistant: No\n",
+    "\"\"\"\n",
+    "End of Examples\n",
+    "\n",
+    "Begin!\n",
+    "\n",
+    "Query: {query}\n",
+    "Retrieved document: {document}\n",
+    "Relevant to the query?\n",
+    "Assistant:\n",
+    "'''\n",
+    "\n",
+    "\n",
+    "@retry(wait=wait_random_exponential(min=1, max=40), stop=stop_after_attempt(3))\n",
+    "def rerank_document(query, document, output_list):\n",
+    "    response = openai.Completion.create(\n",
+    "        model=\"text-davinci-003\",\n",
+    "        prompt=prompt.format(query=query, document=content),\n",
+    "        temperature=0,\n",
+    "        logprobs=1,\n",
+    "        logit_bias={\n",
+    "            5297: 1,\n",
+    "            2949: 1\n",
+    "        }\n",
+    "    )\n",
+    "\n",
+    "    output_list.append(\n",
+    "        (\n",
+    "            query,\n",
+    "            document,\n",
+    "            response[\"choices\"][0][\"text\"],\n",
+    "            response[\"choices\"][0][\"logprobs\"][\"token_logprobs\"][0],\n",
+    "        )\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "753cd363",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "content = result_list[0][\"title\"] + \": \" + result_list[0][\"summary\"]\n",
+    "\n",
+    "# Set logprobs to 1 so we return the most likely token with the likelihood\n",
+    "response = openai.Completion.create(\n",
+    "    model=\"text-davinci-003\",\n",
+    "    prompt=prompt.format(query=query, document=content),\n",
+    "    temperature=0,\n",
+    "    logprobs=1,\n",
+    "    logit_bias={\n",
+    "            5297: 1,\n",
```

Comment:
can go higher than 1 for extra reliability (e.g., 100)

Reply:
and if you want 100% reliability, you can literally append ` Yes` to the end and then just do echo=True and take the last logprob. Guarantees you always get a logprob for ` Yes`. No need to make the change unless you want to.

Reply:
One problem with appending is that you'll get the raw logprob for ` Yes`, which won't just be the ratio of ` Yes` to ` No` but ` Yes` to every other possible token. One nice benefit of the ` Yes` & ` No` upweighting is that you effectively end up with the ratio between them, so you aren't hurt by cases where the model wants to respond with a third thing.

Bottom line: I recommend keeping as is. (Plus, all of these considerations are minor for a task like this which ought to be quite reliable.)

Reply:
Got it, I'll keep as-is for now, but will ask you about this implementation to understand it better.

Reply:
I also tested this implementation and still got a fair number of No's even with a big logit_bias on " Yes" and none on No. Might be worth a discussion.